### PR TITLE
Signup Framework: Step cleanup attempt 2

### PR DESCRIPTION
**Pull request by ddc22**
Related to # https://github.com/Automattic/wp-calypso/pull/80352

## The Issue

* The cleanup done on #80352, removed a step that was being 'phantomly' being submitted on a step. The step was not used in the relevant flow as well.
https://github.com/Automattic/wp-calypso/blob/d36683bdddb147d2e78c69107a2d8373e172a892/client/signup/steps/site-or-domain/index.jsx#L204-L213
* That PR was then reverted in https://github.com/Automattic/wp-calypso/pull/81774 due to the `themeSlugWithRepo` being missing.

## This Fix

* The current PR removes that dead code 
* Provides the `themeSlugWithRepo` properly in the relevent step
* Remerges the unit tests and cleanup from #80352
* and also includes some cautionary comments
* https://github.com/Automattic/wp-calypso/pull/81774

## Testing Instructions

* Make sure the `/start/domain` flow works as expected

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/signup/steps/site-or-domain/index.jsx [hunk 1]
```diff
@@ -201,9 +201,8 @@ class SiteOrDomain extends Component {
 
 		// we can skip the next two steps in the `domain-first` flow if the
 		// user is only purchasing a domain
-		this.props.submitSignupStep( { stepName: 'site-picker', wasSkipped: true } );
 		this.props.submitSignupStep(
-			{ stepName: 'themes', wasSkipped: true },
+			{ stepName: 'site-picker', wasSkipped: true },
 			{ themeSlugWithRepo: 'pub/twentysixteen' }
```
Comment (lsl) [inline client/signup/steps/site-or-domain/index.jsx:206]: Might be wrong here but wondering why we need to keep the themeSlugWithRepo thing, should it not be something like this:

```suggestion
			{ stepName: 'site-picker', wasSkipped: true }
```

Comment (lsl) [reply client/signup/steps/site-or-domain/index.jsx:206]: Or is it because the next step needs that dependency regardless of where it came from?

Comment (ddc22) [reply client/signup/steps/site-or-domain/index.jsx:206]: Yes without the dependency this flow breaks :)

File: client/signup/steps/site-or-domain/index.jsx [hunk 2]
```diff
@@ -223,9 +222,8 @@ class SiteOrDomain extends Component {
 		} else if ( designType === 'existing-site' ) {
```
Comment (southp) [inline client/signup/steps/site-or-domain/index.jsx:222]: Clicking on "Choose site" and then picking any existing site will lead to an infinitely hanging processing screen. I didn't have enough time to dig deeper, but my intuition is that we might need to submit some dependencies here 🤔

**Review by southp — CHANGES_REQUESTED** [review]

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x] Make sure the /start/domain flow works as expected and plan / domain can be purchased at checkout
  - Start a new site option works
  - Use existing site works
  - Just a domain works

### Browsers
- [x] Chrome

File: client/signup/steps/site-picker/site-picker-submit.jsx [hunk 3]
```diff
@@ -14,22 +14,20 @@ export class SitePickerSubmit extends Component {
...
+				siteSlug,
+			},
 			{ themeSlugWithRepo: 'pub/twentysixteen' }
 		);
 
 		if ( hasPaidPlan ) {
 			this.props.submitSignupStep(
 				{ stepName: 'plans-site-selected', wasSkipped: true },
-				{ cartItems: null }
+				{ cartItem: null, themeSlugWithRepo: 'pub/twentysixteen' }
```
Comment (jeyip) [inline client/signup/steps/site-picker/site-picker-submit.jsx]: Wondering if we could explain why we switched this from `cartItems` to `cartItem`? I made the changes a little while ago to accommodate the multiple cart items a plans page might be submitting simultaneously ( storage add-on + plan )

https://github.com/Automattic/wp-calypso/pull/81340/files#diff-4c9407fbe3653d84b6fc1dffdf7c75062e99fa46eb9fdcc62d0b544da8d59eaaR32

I haven't tested the `hasPaidPlan` condition yet, but I'll try and do so when I'm back from lunch

Comment (ddc22) [reply client/signup/steps/site-picker/site-picker-submit.jsx]: > Wondering if we could explain why we switched this from cartItems to cartItem? 

Good catch 💯 
That was unintended!

**Review by southp — APPROVED** [review]
I've spot-checked the affected flows and they all work as expected for me :)

* /start (those use-my-own-domain branches)
* /start/ecommerce
* /start/domain


LGTM. Thanks again for tackling this bold clean-up 🙇🏼

**Pull request merged**